hplsql/src/main/java/org/apache/hive/hplsql/functions/InMemoryFunctionRegistry.java

```diff
@@ -132,39 +133,84 @@ private boolean execProc(String name, HplsqlParser.Expr_func_paramsContext ctx)
+    int actualCnt = (actualValues == null) ? 0 : actualValues.size();
+    int passedParamCnt = actualCnt;
+    List<HplsqlParser.Create_routine_param_itemContext> routineParamItem = formal.create_routine_param_item();
+    int formalCnt = routineParamItem.size();
+    ParserRuleContext ruleContext = (actual == null) ? null : actual.getParent();
+    if (actualCnt > formalCnt) {
+      throw new ArityException(ruleContext, procName, formalCnt, actualCnt);
     }
-    for (int i = 0; i < actualCnt; i++) {
-      HplsqlParser.ExprContext a = actual.func_param(i).expr(); 
-      HplsqlParser.Create_routine_param_itemContext p = getCallParameter(actual, formal, i);
-      String name = p.ident().getText();
-      String type = p.dtype().getText();
-      String len = null;
-      String scale = null;   
-      if (p.dtype_len() != null) {
-        len = p.dtype_len().L_INT(0).getText();
-        if (p.dtype_len().L_INT(1) != null) {
-          scale = p.dtype_len().L_INT(1).getText();
-        }
+    Map<String, Integer> defaultParamNamesVsIndexes = new HashMap<>();
+    if (actualCnt != formalCnt) {
+      populateDefaultParamDetails(routineParamItem, formalCnt, defaultParamNamesVsIndexes);
+      actualCnt = actualCnt + defaultParamNamesVsIndexes.size();
+      if (actualCnt < formalCnt) {
+        throw new ArityException(ruleContext, procName, formalCnt, passedParamCnt);
+      }
+    }
+
+    HplsqlParser.ExprContext exprContext = null;
+    HplsqlParser.Create_routine_param_itemContext paramItemContext = null;
+    Var value = null;
+    // set the passed params
+    for (int i = 0; i < passedParamCnt; i++) {
+      exprContext = actual.func_param(i).expr();
+      paramItemContext = getCallParameter(actual, formal, i);
+      value = actualValues.get(i);
+      // for any default param value is passed then remove it from default param list
+      defaultParamNamesVsIndexes.remove(paramItemContext.ident().getText());
+      setCallParameter(actual, out, exec, exprContext, paramItemContext, value);
+    }
+    // set the remaining default params
+    for (int index : defaultParamNamesVsIndexes.values()) {
+      paramItemContext = formal.create_routine_param_item().get(index);
+      exprContext = paramItemContext.dtype_default().expr();
+      value = exec.evalPop(paramItemContext.dtype_default().expr());
+      setCallParameter(actual, out, exec, exprContext, paramItemContext, value);
+    }
+    // if actual param count + remaining default param count is lesser than formal param count then throw exception as some params are missing
+    if ((passedParamCnt + defaultParamNamesVsIndexes.size()) < formalCnt) {
+      throw new ArityException(ruleContext, procName, formalCnt, passedParamCnt);
+    }
+  }
+
+  private static void populateDefaultParamDetails(List<HplsqlParser.Create_routine_param_itemContext> routineParamItem, int formalCnt,
+      Map<String, Integer> defaultParamNamesVsIndexes) {
+    for (int i = 0; i < formalCnt; i++) {
```

Comment:
how about
```
for (int i = 0; i < routineParamItem.size(); i++)
```
and `formalCnt` can be removed from param list.

Reply:
Fixed